Demangle flags symbols

ret2libc opened the pull request:
<!-- Filling this template is mandatory -->

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [x] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

<!-- Explain the **details** for making this change. Is a new feature implemented? What existing problem does the pull request solve? How does the pull request solve these issues? Please provide enough information so that others can review your pull request. -->

I'd like to have `asm.flags.real=true` by default, however one of the problems with having that enabled is that the users cannot easily know what is the flag name they should use to reference the symbol.

For example:
```
call  std::vector<int, std::allocator<int> >::vector()
```
It is not immediately clear that the flag name (which should be used in commands like `s`) is `method.std::vector_int__std::allocator_int___.vector`.

The idea is that we want the flag name to be displayed as comment (but only when the real names are displayed!):
```
-> asm.flags.real=true

call  std::vector<int, std::allocator<int> >::vector() ; method.std::vector_int__std::allocator_int___.vector
```
```
-> asm.flags.real=false
call  method.std::vector_int__std::allocator_int___.vector
```

These comments are not displayed in visual mode because the user can easily use the "jumpkeys"(the numbers near the calls e.g. `[1]`) to move between functions.

**Test plan**

<!-- What steps should the reviewer take to test your pull request? Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots/videos. This is your time to re-check that everything works and that you covered all the edge cases -->

See attached test.
Moreover, ensure that in visual mode the comments are not displayed.

**Closing issues**

<!-- put "closes #XXXX" in your comment to auto-close the issue that your PR fixes (if any). -->

...

ret2libc committed:
core: do not show realnames in flags if not set

ret2libc committed:
core: show flag name as comments when realnames are displayed

ret2libc committed:
Fix switch/case

XVilka commented:
See also these related bugs: https://github.com/rizinorg/rizin/issues/3656 and https://github.com/rizinorg/rizin/issues/3543

ret2libc committed:
test: fix db/analysis/riscv test

Just fix the test, because the pseudo code seems wrong as well as the
disassembly. `sym.printf` should already be present in the instruction,
thus the flag is redudant.

ret2libc commented:
@XVilka about the test, please see https://github.com/rizinorg/rizin/pull/3915/commits/98f94497c1b2ccfb120eb584e2a4c21999756e88 . Also, i shared some concern in mattermost, but i don't know anything about RISCV.
